python/lsst/ip/isr/isrMockLSST.py

```diff
@@ -0,0 +1,482 @@
+                                               1.0 * self.config.overscanScale)
+
+                # Apply gradient along the Y axis to the image region
+                self.amplifierAddYGradient(ampData, -1.0 * self.config.overscanScale,
+                                           1.0 * self.config.overscanScale)
+
+        # 6. Add Parallel overscan xtalk.
+        # TODO: DM-43286
+
+            if self.config.doAddSerialOverscan:
+                if not self.config.isTrimmed:
+                    # We grow the image to the parallel overscan region
+                    # (we do this instead of using the whole raw region
+                    # in case there are prescan regions)
+                    grownImageBBox = bbox.expandedTo(parallelOscanBBox)
+                    # Now we grow the serial overscan region
+                    # to include the corners
+                    serialOscanBBox = geom.Box2I(
+                        geom.Point2I(serialOscanBBox.getMinX(),
+                                     grownImageBBox.getMinY()),
+                        geom.Extent2I(serialOscanBBox.getWidth(),
+                                      grownImageBBox.getHeight()),
+                    )
+                    serialOscanData = exposure.image[serialOscanBBox]
+
+                    # Add read noise with or without a bias level
+                    # to the serial overscan region.
+                    self.amplifierAddNoise(serialOscanData, self.config.biasLevel
+                                           if self.config.doAddBias else 0.0,
+                                           self.config.readNoise / self.config.gain)
+
+                    # 7. Apply serial overscan in ADU
+                    # Apply gradient along the X axis to both overscan regions.
+                    self.amplifierAddXGradient(serialOscanData, -1.0 * self.config.overscanScale,
+                                               1.0 * self.config.overscanScale)
+                    self.amplifierAddXGradient(parallelOscanData, -1.0 * self.config.overscanScale,
+                                               1.0 * self.config.overscanScale)
+
+                # Apply gradient along the X axis to the image region.
+                self.amplifierAddXGradient(ampData, -1.0 * self.config.overscanScale,
+                                           1.0 * self.config.overscanScale)
+
+        if self.config.doGenerateAmpDict:
+            expDict = dict()
+            for amp in exposure.getDetector():
+                expDict[amp.getName()] = exposure
+            return expDict
+        else:
+            return exposure
+
+    def applyGain(self, ampData, gain):
+        """Apply gain to the amplifier's data.
+
+        Parameters
+        ----------
+        ampData : `lsst.afw.image.ImageF`
+            Amplifier image to operate on.
+        gain : `float`
+            Gain value in e^-/DN.
+        """
+        ampArr = ampData.array
+        ampArr[:] = ampArr[:] / gain
```

Comment:
I'm adding this comment solely to state that this is dividing by the gain, which is correct as part of the mocks, because the mocks need to convert the electron image to ADU (which is the opposite sense than how it's done in IsrTask).  It might be worth adding a note in the docstring, so a code search that finds this will know why it's different than the `isrFunctions.applyGain` method.